File: frontend/devpulse-ui/src/app/core/models/developer-dashboard-dto.ts
export interface DeveloperDashboardDto {
  // summary - Cards
  summaryCardsDto: SummaryCardsDto;

  // Journals Over Time - Line chart
  journalsOverTimeLineChartDto: TimeSeriesPointDto[];

  // Task statuses count - Donut chart
  taskStatusesDonutChartDto: TaskStatusesCountsDto;

  // Feedback vs No Feedback - Bar chart
  journalFeedbackCountsBarChartDto: JournalFeedbackCountsDto;

  // last updated time
  lastUpdated: string;

  // other information
  userId: string;
  userDisplayName: string;
}

export interface TimeSeriesPointDto {
  label: string;
  value: number;
}

export interface SummaryCardsDto {
  highPriorityCount: number;
  newTasksCount: number;
  inProgressTasksCount: number;
  urgentTasksCount: number;
  newFeedbacksCount: number;
}

export interface TaskStatusesCountsDto {
  notStartedTaskCount: number;
  inProgressTaskCount: number;
  completedTaskCount: number;
}

export interface JournalFeedbackCountsDto {
  withFeedBackJournalCount: number;
  withoutFeedBackJournalCount: number;
  totalJounralCount: number;
}

File: backend/SharedLib/DTOs/DeveloperDashboard/DeveloperDashboardDto.cs
using SharedLib.DTOs.ManagerDashboard;

public record DeveloperDashboardDto(
    SummaryCardsDto SummaryCardsDto,
    List<TimeSeriesPointDto> JournalsOverTimeLineChartDto,
    TaskStatusesCountsDto TaskStatusesDonutChartDto,
    JournalFeedbackCountsDto JournalFeedbackCountsBarChartDto,
    string LastUpdated,
    Guid UserId,
    string UserDisplayName
);

public record TimeSeriesPointDto(
    string Label,
    int Value
): LabelNumberDto(Label, Value);

public record SummaryCardsDto(
    int HighPriorityCount,
    int NewTasksCount,
    int InProgressTasksCount,
    int UrgentTasksCount,
    int NewFeedbacksCount
);

public record TaskStatusesCountsDto(
    int NotStartedTaskCount,
    int InProgressTaskCount,
    int CompletedTaskCount
);

public record JournalFeedbackCountsDto(
    int WithFeedBackJournalCount,
    int WithoutFeedBackJournalCount,
    int TotalJounralCount
);